cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DefaultSecureStoreService.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.services;
+
+import co.cask.cdap.api.security.store.SecureStore;
+import co.cask.cdap.api.security.store.SecureStoreData;
+import co.cask.cdap.api.security.store.SecureStoreManager;
+import co.cask.cdap.api.security.store.SecureStoreMetadata;
+import co.cask.cdap.common.BadRequestException;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.SecureKeyId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.proto.security.SecureKeyCreateRequest;
+import co.cask.cdap.proto.security.SecureKeyListEntry;
+import co.cask.cdap.security.authorization.AuthorizerInstantiator;
+import co.cask.cdap.security.spi.authentication.SecurityRequestContext;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Default implementation of the service that manages access to the Secure Store,
+ */
+public class DefaultSecureStoreService implements SecureStoreService {
+  private final AuthorizerInstantiator authorizerInstantiator;
```

Comment:
You can just cache ``Authorizer``, by getting just doing ``authorizerInstantiator.get()`` in the constructor. We had done it this way so far because up until recently, ``AuthorizerInstantiator`` was a guava ``Service`` which would not be started by the time the constructor is injected.